### PR TITLE
Migrate to CSI operator library from library-go

### PR DESCRIPTION
**Pull request by bertinatto**
CC @openshift/storage

### CODE REVIEW

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -3,136 +3,95 @@ package operator
...
+	).WithLogLevelController().WithManagementStateController(
+		operandName,
+		false,
+	).WithStaticResourcesController(
+		"AWSEBSDriverStaticResources",
+		kubeClient,
+		kubeInformersForNamespaces,
+		generated.Asset,
+		[]string{
+			"namespace.yaml", // Is CSO in charge of creating the namespace?
```
Comment (jsafrane) [inline pkg/operator/starter.go:53]: Yes, CSO is in charge. It must create the namespace for the operator.

File: manifests/07_deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        - name: NODE_DRIVER_REGISTRAR_IMAGE
+          value: quay.io/openshift/origin-csi-node-driver-registrar:latest
+        - name: LIVENESS_PROBE_IMAGE
+          value: quay.io/openshift/origin-csi-livenessprobe:latest
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: OPERATOR_NAME
+          value: aws-ebs-csi-driver-operator
```
Comment (jsafrane) [inline manifests/07_deployment.yaml:40]: Is this env. var used  anywhere?

File: manifests/07_deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    metadata:
+      labels:
+        name: aws-ebs-csi-driver-operator
+    spec:
+      containers:
+      - args:
+        - start
+        env:
+        - name: DRIVER_IMAGE
+          value: quay.io/openshift/origin-aws-ebs-csi-driver:latest
```
Comment (jsafrane) [inline manifests/07_deployment.yaml:22]: Not necessarily a bug in this PR, but how we're going to copy this manifest to CSO? We need to add `${DRIVER_IMAGE}`, so CSO can replace it with the current driver image. Can a script in CSO be good enough?
[applies to all images below]

File: manifests/04.rolebinding.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (jsafrane) [inline manifests/04.rolebinding.yaml:1]: the file name `04.rolebinding.yaml` has `.` where other files have `_`

**Comment by bertinatto** [comment]
Failures are not related to this PR:

```
 level=info msg="Creating infrastructure resources..."
level=error
level=error msg="Error: Error launching source instance: InvalidParameterValue: Value (ci-op-9glf012i-79224-8gphp-bootstrap-profile) for parameter iamInstanceProfile.name is invalid. Invalid IAM Instance Profile name"
level=error msg="\tstatus code: 400, request id: 9f813897-5ecb-4742-8180-c910a1fc59ee"
level=error
level=error msg="  on ../tmp/openshift-install-502793661/bootstrap/main.tf line 114, in resource \"aws_instance\" \"bootstrap\":"
level=error msg=" 114: resource \"aws_instance\" \"bootstrap\" {"
level=error
level=error
level=error
level=error msg="Error: Error launching source instance: InvalidParameterValue: Value (ci-op-9glf012i-79224-8gphp-master-profile) for parameter iamInstanceProfile.name is invalid. Invalid IAM Instance Profile name"
level=error msg="\tstatus code: 400, request id: 975c3e7f-c1f9-40b4-abd8-e24a817bba9f"
level=error
level=error msg="  on ../tmp/openshift-install-502793661/master/main.tf line 124, in resource \"aws_instance\" \"master\":"
level=error msg=" 124: resource \"aws_instance\" \"master\" {"
level=error
level=error
level=error
level=error msg="Error: Error launching source instance: InvalidParameterValue: Value (ci-op-9glf012i-79224-8gphp-master-profile) for parameter iamInstanceProfile.name is invalid. Invalid IAM Instance Profile name"
level=error msg="\tstatus code: 400, request id: 5ef19ec8-58c7-4b57-9afe-82747fc7b520"
level=error
level=error msg="  on ../tmp/openshift-install-502793661/master/main.tf line 124, in resource \"aws_instance\" \"master\":"
level=error msg=" 124: resource \"aws_instance\" \"master\" {"
level=error
level=error
level=error
level=error msg="Error: Error launching source instance: InvalidParameterValue: Value (ci-op-9glf012i-79224-8gphp-master-profile) for parameter iamInstanceProfile.name is invalid. Invalid IAM Instance Profile name"
level=error msg="\tstatus code: 400, request id: 40484f5a-8830-405d-a35d-05e9b202fee1"
level=error
level=error msg="  on ../tmp/openshift-install-502793661/master/main.tf line 124, in resource \"aws_instance\" \"master\":"
level=error msg=" 124: resource \"aws_instance\" \"master\" {"
level=error
level=error```

File: manifests/07_deployment.yaml (hunk 2 above)
Comment (bertinatto) [reply manifests/07_deployment.yaml]: It's not. Removed.

File: manifests/04.rolebinding.yaml (hunk 4 above)
Comment (bertinatto) [reply manifests/04.rolebinding.yaml]: Fixed

File: pkg/operator/starter.go (hunk 1 above)
Comment (bertinatto) [reply pkg/operator/starter.go]: Ack

File: manifests/08_cr.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+apiVersion: operator.openshift.io/v1
+kind: "ClusterCSIDriver"
+metadata:
+  name: "ebs.csi.aws.com"
+spec:
+  driverConfig:
+    driverName: "ebs.csi.aws.com"
+  logLevel: Debug
+  managementState: Managed
+  operatorLogLevel: Debug
```
Comment (gnufied) [inline manifests/08_cr.yaml:10]: did we want debug in here?

File: manifests/07_deployment.yaml (hunk 3 above)
Comment (gnufied) [reply manifests/07_deployment.yaml:22]: Do we even need to carry these manifests in this repository? if CSO is going to install them and there is no programming way for CSO to get them, shouldn't this directly live in CSO repo?

File: manifests/08_cr.yaml (hunk 5 above)
Comment (bertinatto) [reply manifests/08_cr.yaml]: nope, just until I get the operator working :)

**Comment by bertinatto** [comment]
/test e2e-operator

>     Get https://api.ci-op-lv624blp-79224.origin-ci-int-aws.dev.rhcloud.com:6443/api/v1/nodes?fieldSelector=spec.unschedulable%3Dfalse&resourceVersion=0: dial tcp 34.193.44.109:6443: connect: connection refused

File: manifests/07_deployment.yaml (hunk 3 above)
Comment (jsafrane) [reply manifests/07_deployment.yaml]: Good point... Who is going to use manifests from this repo? The e2e test will be obsolete as soon as CSO installs the operator by default.

Comment (jsafrane) [reply manifests/07_deployment.yaml]: I'd propose to remove the manifests together with e2e test *after* CSO is updated.

**Comment by jsafrane** [comment]
/retest 
`dial tcp 54.158.134.226:6443: connect: connection refused`

File: manifests/07_deployment.yaml (hunk 3 above)
Comment (jsafrane) [reply manifests/07_deployment.yaml]: We discussed it on a meeting today and we decided to keep the manifests on both places, here and in CSO. We will try to find a better way how to sync the manifests automatically.

File: assets/controller.yaml [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+        app: aws-ebs-csi-driver-controller
+    spec:
+      hostNetwork: true
+      serviceAccount: aws-ebs-csi-driver-controller-sa
+      priorityClassName: system-cluster-critical
+      tolerations:
+        - key: CriticalAddonsOnly
+          operator: Exists
+      containers:
+        # Warning: the operator expects the first container to be the CSI driver
```
Comment (jsafrane) [inline assets/controller.yaml:24]: Is this still true?

File: manifests/04_clusterrolebinding.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: aws-ebs-csi-driver-operator-clusterrole
+subjects:
+  - kind: ServiceAccount
+    name: aws-ebs-csi-driver-operator
+    namespace: openshift-cluster-csi-drivers
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-admin
```
Comment (jsafrane) [inline manifests/04_clusterrolebinding.yaml:12]: This is unfortunate. But we can fix it as a subsequent PR, what do you think?

File: manifests/08_cr.yaml (hunk 5 above)
Comment (bertinatto) [reply manifests/08_cr.yaml]: fixed

File: assets/controller.yaml (hunk 6 above)
Comment (bertinatto) [reply assets/controller.yaml]: no, removed

File: manifests/04_clusterrolebinding.yaml (hunk 7 above)
Comment (bertinatto) [reply manifests/04_clusterrolebinding.yaml]: Fixed

File: manifests/06_clusterrolebinding.yaml [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: aws-ebs-csi-driver-operator-clusterrole
+subjects:
+  - kind: ServiceAccount
+    name: aws-ebs-csi-driver-operator
+    namespace: openshift-cluster-csi-drivers
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-admin
```
Comment (jsafrane) [inline manifests/06_clusterrolebinding.yaml:12]: cluster-admin is still used

File: manifests/05_clusterrole.yaml [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+  - csidrivers
+  verbs:
+  - create
+  - get
+  - list
+  - watch
+  - update
+  - delete
+- apiGroups:
+  - csi.openshift.io
```
Comment (jsafrane) [inline manifests/05_clusterrole.yaml:232]: Is this still needed?

Comment (bertinatto) [reply manifests/05_clusterrole.yaml]: no, removed

File: manifests/06_clusterrolebinding.yaml (hunk 8 above)
Comment (bertinatto) [reply manifests/06_clusterrolebinding.yaml]: fixed

**Pull request merged**